### PR TITLE
Combine registry and image together when registry needs.

### PR DESCRIPTION
**Pull request by hsong-rh**
Both registry and image are needed for docker to pull image from private docker repos.

### CODE REVIEW

**Comment by hsong-rh** [comment]
@Fryguy @bdunne @roliveri @jerryk55 Please review. This should be backported to new G branch.

File: app/models/manageiq/providers/amazon/agent_coordinator.rb [hunk 1]
```diff
@@ -206,7 +206,8 @@ def setup_agent(instance)
     end
 
     # run docker image
-    command_line = "sudo docker run -d --restart=always -v /dev:/host_dev -v #{WORK_DIR}/config.yml:#{WORK_DIR}/config.yml --privileged #{docker_image}"
+    image = docker_login_required? ? "#{docker_registry}/#{docker_image}" : docker_image
```
Comment (roliveri) [inline app/models/manageiq/providers/amazon/agent_coordinator.rb]: @hsong-rh Is `docker_login_required?` the correct test here? I would think we' want to join them whenever `docker_registry` is not empty.

Comment (Fryguy) [reply app/models/manageiq/providers/amazon/agent_coordinator.rb]: Correct, the proper test is when docker_registry is `.present?`

Comment (hsong-rh) [reply app/models/manageiq/providers/amazon/agent_coordinator.rb]: docker_registry is only needed when docker_login_required? is true. I feel checking docker_login_required? is easier to understand.

Comment (bdunne) [reply app/models/manageiq/providers/amazon/agent_coordinator.rb]: > docker_registry is only needed when docker_login_required? is true

That's not true, registries other than docker hub don't always require authentication.

Comment (hsong-rh) [reply app/models/manageiq/providers/amazon/agent_coordinator.rb]: Changed to check docker_registry.present? for combination.

File: app/models/manageiq/providers/amazon/agent_coordinator.rb [hunk 2]
```diff
@@ -206,7 +206,8 @@ def setup_agent(instance)
     end
 
     # run docker image
-    command_line = "sudo docker run -d --restart=always -v /dev:/host_dev -v #{WORK_DIR}/config.yml:#{WORK_DIR}/config.yml --privileged #{docker_image}"
+    image = docker_registry.present? ? "#{docker_registry}/#{docker_image}" : docker_image
```
Comment (bdunne) [inline app/models/manageiq/providers/amazon/agent_coordinator.rb]: Any reason not to use `File.join(docker_registry, docker_image)`?

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit ce1453eb9f7ec77ac8e0fa1c4a164d530b955479
Author: Jason Frey <fryguy9@gmail.com>
Date:   Wed Nov 1 15:15:07 2017 -0400

    Merge pull request #353 from hsong-rh/combine_docker_registry_and_image
    
    Combine registry and image together when registry needs.
    (cherry picked from commit c83987284c6b234437565cdbcce86ef95c5cca60)
```

**Pull request merged**